#2738 locations through form control

joshxg opened the pull request:
Fixes #2738 

## Change summary

- Adds a `FormControl` for `Location` editing. This only has 3 fields - location type, code and description, but if you're just adding a record straight to the list box, it is a lot harder to enforce things - i.e. code, description.
- Adds a column into the Location list box to have a pencil for editing a location.

## Testing

N/A

### Related areas to think about

It's a little naff here because location types are user defined. So you can't really have a list of 'fridges' because the fridge location type would be different for every data file. Also what if you have a type 'large fridge' and 'small fridge' or 'cool room' and 'vaccine fridge'. So you have to list all locations. But then it's in the 'vaccine module' area which is a little odd. 🤷

mark-prins commented on src/localization/formInputStrings.json:
```diff
@@ -7,12 +7,14 @@
     "date_of_birth": "Date of birth",
-    "discount_rate": "Discount rate",
+    "description": "Description",
+    "Discount_rate": "Discount rate",
```
the other ones are still lowercase. I think this one should be too

mark-prins commented:
Am checking now. I tried to add a location and then edit and I got told not to:

![Screen Shot 2020-05-27 at 12 29 51 PM](https://user-images.githubusercontent.com/9192912/82963120-1af1cb80-a016-11ea-8777-c18d8bc0642b.png)

mark-prins commented on src/utilities/formInputConfigs.js:
```diff
@@ -327,6 +348,7 @@ const FORM_CONFIGS = {
     FORM_INPUT_KEYS.SEARCH_FIRST_NAME,
     FORM_INPUT_KEYS.SEARCH_REGISTRATION_CODE,
   ],
+  location: ['code', 'description', 'locationType'],
```
```suggestion
  location: [FORM_INPUT_KEYS.CODE, FORM_INPUT_KEYS.DESCRIPTION, FORM_INPUT_KEYS.LOCATION_TYPE],
```

joshxg commented:
Silly silly me :(

Didn't actually get your error but I am guessing it is because of a lack of `LocationType`s. Have added a few commits - there will be a Toast now when you try to add or edit a location when you don't have any location types. I was thinking of having code in msupply to always ensure there is at least one location type (with like.. SUSSOL_FRIDGE?) or something could be an idea.. but I'm not sure. This is difficult with it being user defined !

joshxg commented:
Dumb, I can't reply to your comments  ! :(

Removed from UIDatabase: 953eaef46180676008ba8c1d4192907fc75c3d8d
Reverted the capitals mishap: fb6c8062f3572a7fb748a79ea7d80ae681bc37f7
use constants: 28447732abaf484754cf3693ff64761b94f5dcc5

Thanks!!
